docs/changelog.md

```diff
@@ -26,6 +26,9 @@ permalink: /changelog/
 
 * **support-webextensions**
   * 🆕 New component containing building blocks for features implemented as web extensions.
+  
+* **service-glean**
+  * Glean now registers its lifecycle observers on the main thread.
```

Comment:
For future reference, any chance you could link to the bug that started it all (bug 1581556) and to the [google issue](https://issuetracker.google.com/issues/138953075#comment2) that explicitly says this is expected as registration must happen on the main thread?

Maybe the bug number is fine, since you stated all the details in the PR message and we can always trace back to it.